packages/grid/src/vaadin-grid-data-provider-mixin.js

```diff
@@ -204,26 +204,34 @@ export const DataProviderMixin = (superClass) =>
      * @param {HTMLElement} row
      * @protected
      */
-    _getItem(row) {
+    _requestOrUpdateRowItem(row) {
```

Comment:
not 100% safe to rename protected methods. I'd assume it's fine in this case.

Reply:
True, but I've assumed that it shouldn't be a big deal since the change is overall quite minor and this method isn't overridden in the connector. If any custom extensions happen to use it, it shouldn't be difficult to rename it on their side. Would you like it to be marked as a breaking change though?

Reply:
Not sure why this method needs to be protected in general.

Reply:
> Would you prefer it to be marked as a breaking change though?

No, I assume there's no external use for this method. It could even be changed to `private` by this PR.

Reply:
Let me make it private.

Reply:
Oh, I take that back https://github.com/vaadin-component-factory/selection-grid-flow/blob/770bab58e6ef1457ab6e948d27705fbfe9cfbb1a/selection-grid-flow/src/main/resources/META-INF/resources/frontend/src/vcf-selection-grid.js#L51

Reply:
Oh, ok. Do you think it's worth deprecating then? Or marking as a breaking change would be enough?

Reply:
However, its deprecation will require us to use it everywhere instead of the new method, making the value of the PR questionable :)

Reply:
Not sure, seems that the SelectionGrid addon [doesn't use/support V24 version](https://vaadin.com/directory/component/selection-grid) of the grid anyway so it shouldn't be affected

Reply:
It seems to only have experimental support of V24: https://github.com/vaadin-component-factory/selection-grid-flow/commit/6f6521fe03b24a70f80fc967587cabc25a88a772

Reply:
Well, I'm leaning toward marking the PR as a breaking change with an explanation in the description. Ultimately, this change isn't as big as the introduction of DataProviderController which required some deprecations.

Reply:
Note, there is also `_updateItem` which could be renamed to `_updateRowItem` for consistency but I decided to not touch it because it's also used in the connector.